nemo/collections/nlp/models/language_modeling/transformer_lm_model.py

```diff
@@ -163,16 +184,49 @@ def setup_test_data(self, test_data_config: Optional[DictConfig]):
         )
 
     def _setup_dataloader_from_config(self, cfg: DictConfig, predict_last_k=0):
-        dataset = L2RLanguageModelingDataset(
-            tokenizer=self.tokenizer,
-            dataset=cfg.file_name,
-            max_seq_length=self.dataset_cfg.max_seq_length,
-            batch_step=predict_last_k,
-        )
+        if cfg.get('cache_ids', False):
+            logging.info("Constructing tokenized dataset cache...")
+
+        shuffle = cfg.shuffle
+
+        if cfg.get('is_tarred', False):
+            if ('tarred_text_filepaths' in cfg and cfg['tarred_text_filepaths'] is None) or (
+                'file_name' in cfg and cfg['file_name'] is None
+            ):
+                logging.warning(
+                    "Could not load dataset as `file_name` was None or "
+                    f"`tarred_text_filepaths` is None. Provided config : {cfg}"
+                )
+                return None
+
+            shuffle_n = cfg.get('shuffle_n', 4 * cfg['batch_size']) if shuffle else 0
```

Comment:
Do you think we can go higher than 4 times here?

Reply:
We can, this is just the default used for ASR. Technically speaking, we have a very small amount of tarfiles (64) for nlp compared to ASR (512/1024) + each tarfile contains very few chunks of samples (75-80 .npy files). So practically it doesnt matter what shuffle size you put, as long as its over 100, the entire tarfile will get shuffled for small amounts of text. Its only when we go to TB of text that this value beomes somewhat important.